base/range.jl

```diff
@@ -861,13 +861,15 @@ end
 median(r::Range{<:Real}) = mean(r)
 
 function in(x, r::Range)
+    applicable(-, x, first(r)) || return false
     n = step(r) == 0 ? 1 : round(Integer,(x-first(r))/step(r))+1
```

Comment:
Mmm. You're right that when the condition is true, the rest of the body isn't optimized out as I would have expected. So `n`  still exists and is inferred as `Any`; though the return value is correctly inferred as `Bool`.

More fundamentally, `applicable`/`method_exists` does not appear to be resolved at compile time. That's because https://github.com/JuliaLang/julia/pull/16422 hasn't been merged (I had forgotten about this). So I have no solution for now...